src/main/java/com/igot/cb/demand/service/DemandServiceImpl.java

```diff
@@ -122,7 +122,7 @@ public CustomResponse createDemand(JsonNode demandDetails, String token, String
                 ((ObjectNode) demandDetails).put(Constants.CREATED_ON, String.valueOf(currentTime));
                 jsonNodeEntity.setCreatedOn(currentTime);
             }
-            if (!demandDetails.get(Constants.ASSIGNED_PROVIDER).isEmpty()){
+            if (demandDetails.has(Constants.DEMAND_ID) && demandDetails.get(Constants.REQUEST_TYPE).asText().equalsIgnoreCase(Constants.SINGLE)){
```

Comment:
Always it must be --> constant.equalsIgnoreCase(variable)

Constants.SINGLE.equalsIgnoreCase(demandDetails.get(Constants.REQUEST_TYPE))

Reply:
Resolved